src/main/java/pl/tomaszdziurko/jvm_bloggers/view/login/attack/stream/BruteForceAttackEventStreamManager.java

```diff
@@ -0,0 +1,90 @@
+package pl.tomaszdziurko.jvm_bloggers.view.login.attack.stream;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.tomaszdziurko.jvm_bloggers.mailing.MailSender;
+import pl.tomaszdziurko.jvm_bloggers.view.login.attack.BruteForceAttackEvent;
+import pl.tomaszdziurko.jvm_bloggers.view.login.attack.BruteForceAttackMailGenerator;
+import rx.Observable;
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+import rx.subjects.PublishSubject;
+
+import javax.annotation.PreDestroy;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * For field based DI say big "Thank You" to Apache Wicket ;)
```

Comment:
Then try to postpone creating PR if your code is not in final state yet. It will save precious time of the reviewers :)